### PR TITLE
[20392] Remove fixed_size_string file in fastrtps

### PR DESCRIPTION
**Pull request by elianalf**
## Description

This pr removes fastrtps/utils/fixed_size_string.hpp and substitutes it with fastcdr/cdr/fixed_size_string.hpp.
Related documentation PR https://github.com/eProsima/Fast-DDS-docs/pull/695 .

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- N/A Changes are ABI compatible. 
- N/A Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/unittest/utils/FixedSizeStringTests.cpp
Comment (JLBuenoLopez-eProsima) [inline test/unittest/utils/FixedSizeStringTests.cpp:1]: This file was migrated to Fast CDR so these tests can be removed from Fast DDS.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: include/fastdds/dds/builtin/common/Types.hpp [hunk 1]
```diff
@@ -34,7 +32,7 @@ namespace rpc {
 
 using InstanceName =
 #if FASTCDR_VERSION_MAJOR == 1
-        fastrtps::string_255;
+        fastcdr::string_255;
 #else
         fastcdr::string_255;
 #endif // FASTCDR_VERSION_MAJOR == 1
```
[the remark this answers is not in the document]
Comment (elianalf) [reply include/fastdds/dds/builtin/common/Types.hpp:37]: Absolutely, good idea! But since it is also related to other header files and part of the code, I would open a different pr. Let me know what you think

[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply include/fastdds/dds/builtin/common/Types.hpp:37]: My intention was to just remove it in this specific place. But it is true that we should remove it also in other places and that should be done in a specific PR.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
This branch is not building because it is still including the removed source file `FixedSizeStringTests.cpp` file

**Comment by elianalf** [comment]
@richiprosima please test this

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Comment by elianalf** [comment]
@richiprosima please test_3 this

File: include/fastdds/rtps/builtin/data/WriterProxyData.h [hunk 2]
```diff
@@ -29,7 +31,7 @@
 #include <fastdds/rtps/security/accesscontrol/EndpointSecurityAttributes.h>
 #endif // if HAVE_SECURITY
 #include <fastrtps/attributes/TopicAttributes.h>
-#include <fastrtps/utils/fixed_size_string.hpp>
+#include <fastcdr/cdr/fixed_size_string.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline include/fastdds/rtps/builtin/data/WriterProxyData.h:34]: This header is repeated

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp [hunk 3]
```diff
@@ -18,7 +18,6 @@
  */
 
 #include <fastdds/rtps/common/CacheChange.h>
-#include <fastrtps/utils/fixed_size_string.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp:21]: The header is being removed without updating. And the fixed size string API is being used in this file.

File: src/cpp/rtps/builtin/discovery/endpoint/EDPServer.cpp [hunk 4]
```diff
@@ -16,6 +16,8 @@
  * @file EDPServer.cpp
  *
  */
+#include <rtps/builtin/discovery/endpoint/EDPServer.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/builtin/discovery/endpoint/EDPServer.cpp:19]: Only the corresponding header file should be the first. Other headers should keep the expected order. This is also happening in other files. Please, revert this change, leaving only the specific header file corresponding to the source file.

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
Some nitpicks

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM with green CI

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Comment by elianalf** [comment]
@richiprosima please test_3 this
